fuse_constraints/include/fuse_constraints/fixed_3d_landmark_simple_covariance_constraint.h

```diff
@@ -0,0 +1,229 @@
+   * @param[in] mean          The measured/prior pose of the marker as a vector (7x1 vector: x, y, z, qw, qx, qy, qz)
+   * @param[in] covariance    The detection covariance, in pixels (2x2 matrix: u, v)
+   */
+  Fixed3DLandmarkSimpleCovarianceConstraint(const std::string& source,
+                                            const fuse_variables::Position3DStamped& position,
+                                            const fuse_variables::Orientation3DStamped& orientation,
+                                            const fuse_variables::PinholeCamera& calibraton,
+                                            const fuse_core::MatrixXd& pts3d, const fuse_core::MatrixXd& observations,
+                                            const fuse_core::Vector7d& mean, const fuse_core::Matrix2d& covariance);
+
+  /**
+   * @brief Create a constraint using a known 3D fiducial marker. Convenience constructor for the special case of
+   *        an ARTag with 4 corners.
+   *
+   * @param[in] source        The name of the sensor or motion model that generated this constraint
+   * @param[in] position      The variable representing the position components of the marker pose
+   * @param[in] orientation   The variable representing the orientation components of the marker pose
+   * @param[in] calibraton    The intrinsic calibration parameters of the camera (4x1 vector: cx, cy, fx, fy).
+   *                          NOTE: Best practice is to fix this variable unless we have several observations
+   *                          with the same camera
+   * @param[in] marker_size   The size of the marker, in meters. Assumed to be square.
+   * @param[in] observations  The 2D (pixel) observations of each marker's corners.
+   * @param[in] mean          The measured/prior pose of the marker as a vector (7x1 vector: x, y, z, qw, qx, qy, qz)
+   * @param[in] covariance    The detection covariance, in pixels (2x2 matrix: u, v)
+   */
+  Fixed3DLandmarkSimpleCovarianceConstraint(const std::string& source,
+                                            const fuse_variables::Position3DStamped& position,
+                                            const fuse_variables::Orientation3DStamped& orientation,
+                                            const fuse_variables::PinholeCamera& calibraton, const double& marker_size,
+                                            const fuse_core::MatrixXd& observations, const fuse_core::Vector7d& mean,
+                                            const fuse_core::Matrix2d& covariance);
+
+  /**
+   * @brief Destructor
+   */
+  virtual ~Fixed3DLandmarkSimpleCovarianceConstraint() = default;
+
+  /**
+   * @brief Read-only access to the measured/prior vector of mean values.
+   *
+   * Order is (x, y, z, qw, qx, qy, qz)
+   */
+  const fuse_core::Vector7d& mean() const
+  {
+    return mean_;
+  }
+
+  /**
+   * @brief Read-only access to the square root information matrix.
+   *
+   * Order is (x, y, z, qx, qy, qz)
```

Comment:
The covariance/information matrix is only 2x2. The order should be (x, y) or (u, v), correct?
Same with the `covariance()` method below.

Reply:
Correct, this case is the "simple" covariance (i.e. pixel error)